var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -153,24 +153,22 @@ def _install_binaries(self, mech_name=None):
 
         if (self.spec.satisfies('^neuron~binary') or
                 self.spec.satisfies('^neuron+binary+cmake')):
-            lib_suffix = ".so"
+            if 'darwin' in self.spec.architecture:
+                dso_suffix = '.dylib'
+            else:
+                dso_suffix = '.so'
             # Install libnrnmech - might have several links
             if self.spec.satisfies('^neuron+cmake'):
                 libnrnmech_path = self.nrnivmodl_outdir
-                if "darwin" in self.spec.architecture:
-                    lib_suffix = ".dylib"
             else:
-                libnrnmech_path = self.nrnivmodl_outdir + "/.libs"
+                libnrnmech_path = self.nrnivmodl_outdir + '/.libs'
             for f in find(libnrnmech_path,
-                          'libnrnmech*' + lib_suffix + '*',
+                          'libnrnmech*' + dso_suffix + '*',
```

Comment:
again, not sure we need to get the extension if we can use a dot to be the limit of the lib name

Reply:
Probably this should work here as well. I will test to make sure it works fine and change it